dev/core/src/com/google/gwt/dev/jjs/ast/JCaseStatement.java

```diff
@@ -17,28 +17,68 @@
 
 import com.google.gwt.dev.jjs.SourceInfo;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Java case statement.
  */
 public class JCaseStatement extends JStatement {
 
-  private JExpression expr;
+  private List<JExpression> exprs;
 
   public JCaseStatement(SourceInfo info, JExpression expr) {
     super(info);
-    this.expr = expr;
+    this.exprs = Collections.singletonList(expr);
+  }
+
+  public JCaseStatement(SourceInfo info, Collection<JExpression> exprs) {
+    super(info);
+    this.exprs = Collections.unmodifiableList(new ArrayList<>(exprs));
   }
 
+  public boolean isDefault() {
+    return exprs.isEmpty();
+  }
+
+  @Deprecated
   public JExpression getExpr() {
-    return expr;
+    if (exprs.size() > 1) {
+      throw new IllegalStateException(
+              "JCaseStatement.getExpr() called on a node with multiple expressions " + exprs);
+    }
+    return exprs.isEmpty() ? null : exprs.get(0);
+  }
+
+  public List<JExpression> getExprs() {
+    return exprs;
+  }
+
+  public JBinaryOperation convertToCompareExpression(JExpression value) {
+    if (isDefault()) {
+      throw new IllegalStateException("Can't replace a default statement with a comparison");
+    }
+    JBinaryOperation compareOperation = null;
+    for (JExpression expr : getExprs()) {
+      JBinaryOperation caseComparison = new JBinaryOperation(getSourceInfo(),
+              JPrimitiveType.BOOLEAN, JBinaryOperator.EQ, value, expr);
+      if (compareOperation == null) {
+        compareOperation = caseComparison;
+      } else {
+        compareOperation = new JBinaryOperation(getSourceInfo(), JPrimitiveType.BOOLEAN,
+                JBinaryOperator.OR, compareOperation, caseComparison);
+      }
+    }
+    assert compareOperation != null : this;
+    return compareOperation;
   }
```

Comment:
Except for the default check - which I think is a guard since the places where this is called we should never get here - this is method seems to be generic and not specific to the switch case statements, So I am not sure if this the best place for it, seems like it belong somewhere in the GwtAstBuilder. WDYT?

Reply:
I only moved it here because I ended up having the logic in several places, or calling into some place like GwtAstBuilder.

The GwtAstBuilder type is intended for converting JDT ast into GWT's Java ast - but we only use this in code that optimizes out the GWT Java ast - DeadCodeElimination and CfgBuilder.

I'll Javadoc it so it is more clear, but this is definitely specific to a CaseStatement - the `getExprs()` call in the loop is what this is about. The goal is to rewrite a single `case` into an `if` so we can remove the `switch` entirely, which applies when optimizing code (after some case blocks are optimized out or enum values are removed, etc).

I don't see an obvious class to put this in as a static method, do you?